### PR TITLE
Add Implementation Details for the project

### PR DESCRIPTION
**Pull request by benardop**
Changes I have made in this PR:
-  Add implementation script files for all components of the application
- Add styling for each component
- Add tests scripts for each component in the application

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,58 @@
-# Getting Started with Create React App
+# Country-Facts
+Microverse Module 03 Project: This educational project is to create a React Country-Facts Application that allows the user to display various metric information about a given selected Country from an API.
+
```
Comment (mikethreels) [inline README.md]: - [ ] I'm not able to find a link to your presentation video please add it before requesting another review.

File: README.md [hunk 2]
```diff
@@ -1,70 +1,58 @@
...
+## Show your support
 
-Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you're on your own.
+Give a ⭐ if you like this project!
 
-You don't have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn't feel obligated to use this feature. However we understand that this tool wouldn't be useful if you couldn't customize it when you are ready for it.
+## Acknowledgments
 
-## Learn More
+- Thanks to Microverse team and my session Coding Partners
```
Comment (mikethreels) [inline README.md]: - [ ] The README should contain an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: README.md [hunk 3]
```diff
@@ -1,70 +1,58 @@
...
+- React.js
+- Javascript
+- JSX
+- Babel
+- Redux
+- Linters
 
-## Available Scripts
+## Live Demo
+[Live Demo Link](Coming soon)
```
Comment (mikethreels) [inline README.md]: - [ ] you should according to the requirements deploy the application using Heroku, Netlify or GitHub Pages.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Bernard, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 4]
```diff
@@ -1,70 +1,69 @@
...
+- CSS
+- React.js
+- Javascript
+- Babel
+- Redux
+- Linters
 
-## Available Scripts
+## Live Demo
+[Live Demo Link](https://benardop.github.io/country-facts/)
```
Comment (TedLivist) [inline README.md:13]: - [ ] Kindly ensure your app is properly deployed. When I visit the site, it displays nothing. Kindly fix that.

  fkkgi

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @benardop,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Benard, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions of the previous CR.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**